pages/price-feeds/use-real-time-data/solana.mdx

```diff
@@ -1,23 +1,149 @@
+Add this crate to the dependencies section of your `Cargo.toml` file:
 
-- [Devnet](https://pyth.network/developers/price-feed-ids#solana-devnet)
-- [Testnet](https://pyth.network/developers/price-feed-ids#solana-testnet)
-- [Mainnet-Beta](https://pyth.network/developers/price-feed-ids#solana-mainnet-beta)
+```toml copy
+[dependencies]
+pyth-solana-receiver-sdk ="0.1.0"
+```
 
-## Program Addresses
+### Typescript SDK
 
-Developers will need the address of the Pyth oracle πrogram on their blockchain in order to use Pyth.
-Please consult [Solana Contract Addresses](../contract-addresses/solana) to find the address for your blockchain.
+Pyth provides two Typescript packages, [@pythnetwork/price-service-client](https://github.com/pyth-network/pyth-crosschain/tree/main/price_service/client/js) and [@pythnetwork/pyth-solana-receiver](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/solana/sdk/js/pyth_solana_receiver), for fetching Pyth prices and submitting them to the blockchain respectively.
+Add these packages to your off-chain dependencies:
+
+```bash copy
+npm install --save @pythnetwork/price-service-client @pythnetwork/pyth-solana-receiver
+```
+
+## Write Contract Code
+
+Add code to your Solana program to read the Pyth price.
+Pyth prices are posted to price update accounts that can be passed to any instruction that needs price data.
+If you are using Anchor, you can simply add an `Account<'info, PriceUpdateV2>` field to your `Context` struct:
+
+TODO: does this Account thing perform the necessary account ownership checks?
+
+```rust copy
+#[derive(Accounts)]
+#[instruction()]
+pub struct Sample<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    // Add this account to any instruction Context that needs price data.
+    pub price_update: Account<'info, PriceUpdateV2>,
+}
+```
+
+Next, update the instruction logic to read the price from the price update account:
+
+```rust copy
+pub fn sample(ctx: Context<Sample>) -> Result<()> {
+    let price_update = &mut ctx.accounts.price_update;
+    // get_price_no_older_than will fail if the price update is more than 30 seconds old
+    let maximum_age: u64 = 30;
+    // get_price_no_older_than will fail if the price update is for a different price feed.
+    // This string is the id of the BTC/USD feed. See https://pyth.network/developers/price-feed-ids for all available ids.
+    let feed_id: [u8; 32] = get_feed_id_from_hex("0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43")?;
+    let price = price_update.get_price_no_older_than(&Clock::get()?, maximum_age, &feed_id)?;
+    // Sample output:
+    // The price is (7160106530699 ± 5129162301) * 10^-8
+    msg!("The price is ({} ± {}) * 10^{}", price.price, price.conf, price.exponent);
+
+    Ok(())
+}
+```
+
+Note that it is your responsibility in this function to validate that the provided price update is for the appropriate price feed and timestamp.
```

Comment:
This can we written something as `Caution: We recommend users to validate price update matches the respective price feed and timestamp`

Reply:
I think it needs to be more strongly worded than that -- this is a serious security issue if they forget it